packages/cells/src/widget.ts

```diff
@@ -1945,9 +1945,9 @@ export class MarkdownCell extends AttachmentsCell<IMarkdownCellModel> {
   }
 
   /**
-   * Renderer
+   * Markdown rendered container
    */
-  get renderer(): IRenderMime.IRenderer {
+  get renderer(): Widget {
```

Comment:
Why was the type changed to `Widget` rather than `IRenderMime.IRenderer` as it was? It seems that `IRenderer` inherits from `Widget` so at first glance it should be better as-is?

https://github.com/jupyterlab/jupyterlab/blob/552cdfe9314365d668c029e0863cf2fdbbbee4df/packages/rendermime-interfaces/src/index.ts#L288C20-L301

Reply:
Reusing the rendermime renderer for the markdown cell is an implementation detail (actually we need to build a rendermime datamodel each time just to use it). When extending the markdown class, the only interest is to provide a Widget to be injected in the input area via:

https://github.com/jupyterlab/jupyterlab/blob/ecec3f6c90f54c4ad41b922e5ab16b3cae321953/packages/cells/src/inputarea.ts#L112

That method already accept a more generic `Widget`.

Reply:
The public getter was added in 

https://github.com/jupyterlab/jupyterlab/pull/11689/files#diff-352916c4cc928b1298618fffc0424a71ee8816925f3a36c77d8593057e7b6f89

because it is needed to search in the rendered markdown widget at (`GenericSearchProvider` expect a `Widget`):

https://github.com/jupyterlab/jupyterlab/blob/ecec3f6c90f54c4ad41b922e5ab16b3cae321953/packages/cells/src/searchprovider.ts#L579-L581

There are no other usage and I exported the more restrictive `IRenderMime.IRenderer` as it was the type of the variable.

Reply:
https://github.com/jupyterlab/jupyterlab/blob/21820b8762d9f058defe4411d9ad77c7e3d558a9/packages/cells/src/widget.ts#L2166

What I was worried about is that downstream extension authors could override renderer getter with a `Widget` without a `renderModel` method which would break the above. Is my reasoning correct?

Reply:
Yes you are right